### PR TITLE
Various code cleanups

### PR DESCRIPTION
**Pull request by tamirkamara**
The following changes are proposed:

* Using generic net() rather than the explicit type
* Upgrade analyzer package version
* Remove obsolete suppressions
* Fix various lint errors

### CODE REVIEW

File: K2Bridge/K2Bridge.csproj [hunk 1]
```diff
@@ -16,11 +16,11 @@
     <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.1.2" />
     <PackageReference Include="Microsoft.AspNetCore.Mvc.WebApiCompatShim" Version="2.2.0" />
     <PackageReference Include="Microsoft.Azure.Kusto.Data" Version="9.2.0" />
-    <PackageReference Include="System.Data.SqlClient" Version="4.8.2" />
-    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.9.8">
+    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="6.0.0">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
+    <PackageReference Include="System.Data.SqlClient" Version="4.8.2" />
```
Comment (eladiw) [inline K2Bridge/K2Bridge.csproj:23]: why we need this?

Comment (tamirkamara) [reply K2Bridge/K2Bridge.csproj:23]: I don't know but it was here before...

Comment (eladiw) [reply K2Bridge/K2Bridge.csproj:23]: moved from above (:

**Review by eladiw — APPROVED** [review]
lgtm

File: K2Bridge/K2Exception.cs [hunk 2]
```diff
@@ -52,6 +52,7 @@ public K2Exception(string message, Exception innerException)
         /// <param name="message">The message that describes the error.</param>
         /// <param name="innerException">The exception that is the cause of the current exception,
         /// or a null reference(Nothing in Visual Basic) if no inner exception is specified.</param>
+        /// <param name="phaseName">The logical phase where the exception originated from.</param>
```
Comment (AsafMah) [inline K2Bridge/K2Exception.cs:55]: Why did you remove this doc?

File: K2Bridge/KustoDAL/HitsMapper.cs [hunk 3]
```diff
@@ -23,7 +23,7 @@ public static class HitsMapper
         /// <summary>
         /// Type converter function used to get values from column.
         /// </summary>
-        private static readonly Dictionary<Type, Func<object, object>> Converters = new Dictionary<Type, Func<object, object>>
```
Comment (AsafMah) [inline K2Bridge/KustoDAL/HitsMapper.cs:26]: If you're modernizing the code, consider using the modern dict syntax:
```
        private static readonly Dictionary<Type, Func<object, object>> Converters = new ()
        {
            [typeof(sbyte)] = value => value is DBNull || value == null ? null : (bool?)((sbyte)value != 0),
            [typeof(SqlDecimal)] = value => value.Equals(SqlDecimal.Null) ? double.NaN : ((SqlDecimal)value).ToDouble(),
            [typeof(Guid)] = value => value is DBNull || value == null ? null : ((Guid)value).ToString(),
            [typeof(TimeSpan)] = value => value is DBNull || value == null ? null : XmlConvert.ToString((TimeSpan)value),

            // Elasticsearch returns timestamp fields in UTC in ISO-8601 but without Timezone.
            // Use a String type to control serialization to mimic this behavior.
            [typeof(DateTime)] = value => value is DBNull ? null : ((DateTime)value).ToString("yyyy-MM-dd'T'HH:mm:ss.FFFFFFF"),
        };
```
Or even further with the "or null" pattern:

```
        private static readonly Dictionary<Type, Func<object, object>> Converters = new ()
        {
            [typeof(sbyte)] = value => value is DBNull or null ? null : (bool?)((sbyte)value != 0),
            [typeof(SqlDecimal)] = value => value.Equals(SqlDecimal.Null) ? double.NaN : ((SqlDecimal)value).ToDouble(),
            [typeof(Guid)] = value => value is DBNull or null ? null : ((Guid)value).ToString(),
            [typeof(TimeSpan)] = value => value is DBNull or null ? null : XmlConvert.ToString((TimeSpan)value),

            // Elasticsearch returns timestamp fields in UTC in ISO-8601 but without Timezone.
            // Use a String type to control serialization to mimic this behavior.
            [typeof(DateTime)] = value => value is DBNull ? null : ((DateTime)value).ToString("yyyy-MM-dd'T'HH:mm:ss.FFFFFFF"),
        };
```

**Review by AsafMah — APPROVED** [review]

**Review by tomconte — APPROVED** [review]

File: K2Bridge/KustoDAL/HitsMapper.cs (hunk 3 above)
Comment (tamirkamara) [reply K2Bridge/KustoDAL/HitsMapper.cs:26]: thank you @AsafMah - it's done.
